cpu/samd5x/include/periph_cpu.h

```diff
@@ -83,6 +83,19 @@ enum {
  */
 #define SPI_HWCS(x)     (UINT_MAX - 1)
 
+#ifndef DOXYGEN
+#define HAVE_ADC_RES_T
+typedef enum {
+    ADC_RES_6BIT  = 0xff,                       /**< not supported */
+    ADC_RES_8BIT  = ADC_CTRLB_RESSEL_8BIT,      /**< ADC resolution: 8 bit */
+    ADC_RES_10BIT = ADC_CTRLB_RESSEL_10BIT,     /**< ADC resolution: 10 bit */
+    ADC_RES_12BIT = ADC_CTRLB_RESSEL_12BIT,     /**< ADC resolution: 12 bit */
+    ADC_RES_14BIT = 0xfe,                       /**< not supported */
+    ADC_RES_16BIT = 0xfd                        /**< not supported */
+} adc_res_t;
+/** @} */
+#endif /* DOXYGEN */
```

Comment:
You could add a

```C
#define ADC_GCLK_SRC_DEFAULT  SAM0_GCLK_TIMER
```

I don't think the ADC will run off a 120 MHz clock.

Reply:
Ah I see you have a prescaler of 256 in the board config.
So the ADC is running at 468 kHz which albeit a bit slow, should be a valid configuration.

Reply:
I'll lower the prescaler to 128. Is a bit faster and still runs.